R/mergeSEs.R

```diff
@@ -491,30 +491,10 @@ setMethod("mergeSEs", signature = c(x = "list"),
                 "is discarded.", call. = FALSE)
         return(tse)
     }
-    # If there are multiple trees, select non-duplicated trees; the largest
-    # take the precedence, remove duplicated rowlinks --> each row is presented
-    # in the set only once --> remove trees that do not have any values anymore.
-    # The aim is to subset the dataset so that it is easier to handle in tree
-    # binding step for instance. Otherwise, it would lead to huge tree that
-    # might exceed memory.
-    if( length(trees) > 1 ){
-        # Sort trees --> trees with highest number of taxa first
-        max_trees <- table(links$whichTree)
-        max_trees <- names(max_trees)[order(max_trees, decreasing = TRUE)]
-        # Order the link data frame, take largest trees first
-        links$whichTree <- factor(links$whichTree, levels = max_trees)
-        links <- links[order(links$whichTree), ]
-        # Remove factorization
-        links$whichTree <- unfactor(links$whichTree)
-        # Remove duplicated links
-        links <- links[!duplicated(links$names), ]
-        # Subset trees
-        trees <- trees[unique(links$whichTree)]
-    }
```

Comment:
I noticed that this might lead to problems if the trees are named equally. This can happen in `unsplitOn()` function. This chunk was not necessarily need, it was just to remove duplicated trees. After removing this, the memory issue is fixed later.